ddtrace/debugging/config.py

```diff
@@ -0,0 +1,111 @@
+import os
+from typing import Dict
+from typing import Optional
+
+from ddtrace import config as tracer_config
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.utils.config import get_application_name
+from ddtrace.internal.utils.formats import asbool
+from ddtrace.internal.utils.formats import parse_tags_str
+from ddtrace.version import get_version
+
+
+log = get_logger(__name__)
+
+
+DD_SITE_MAP = {"datad0g.com": "https://dd.datad0g.com"}
+
+DEFAULT_DEBUGGER_PORT = 8126
+DEFAULT_PROBE_API_URL = "https://app.datadoghq.com"
+DEFAULT_SNAPSHOT_INTAKE_URL = "http://%s:%d" % (os.getenv("DD_AGENT_HOST", "localhost"), DEFAULT_DEBUGGER_PORT)
+DEFAULT_SERVICE_NAME = "unnamed-python-service"
+DEFAULT_MAX_PROBES = 100
+DEFAULT_METRICS = True
+DEFAULT_PROBE_STATUS_URL = DEFAULT_SNAPSHOT_INTAKE_URL
+DEFAULT_GLOBAL_RATE_LIMIT = 100.0
+DEFAULT_AGENT_MODE = False
+DEFAULT_API_KEY_FILE = "/etc/datadog/secrets/datadog-api-key"
+
+
+def _get_api_key_from_file():
+    # type: () -> Optional[str]
+    """Retrieve the API key from file."""
+    try:
+        key_file = os.getenv("DD_API_KEY_FILE", DEFAULT_API_KEY_FILE)
+        with open(key_file) as _:
+            return _.read().strip()
+    except (IOError, OSError):
+        return None
+
+
+SECRETS = ["api_key"]
+
+
+class _DebuggerConfig(object):
```

Comment:
Are run-time mutations to the attributes of this class to be considered by the debugger?

If so can we explicitly document that this is the expected behaviour? If not then can we privatize all attributes on the object so it doesn't appear that it can be changed at run-time?

We have this problem with the existing global/integration configs in that it's not clear when they can be mutated and if it'll have an effect or not which (IMO is super frustrating/confusing from a user perspective).

Reply:
Good point. Mutating the configuration object has no effect while a debugger is enabled. For changes to take effect, the users would have to `disable()` and then `enable()`. But whilst this would work, I don't think we want to encourage this and quite likely there shouldn't be any need for doing this, as most of the dynamic configuration is coming from other sources.

In the end, the configuration object is supposed to be an internal interface around the environment, so it shouldn't be acted upon by users.